SparkPost/2web2ui
# AC-1937 - Fix form submit events in the places it is not triggering

AC-1937 - Fix form submit events in the places it is not triggering

### What Changed

- In BackupCodeModal, moved the onClick from Button to Form's handleSubmit and updated the unit test accordingly.
- In EnableTfaForm and DisableTfa form made the same change as above.
-----------------------------------------------------------------------------
- In InsertSnippetModal, removed the onClick from submit button and updated the test accordingly

### How To Test
Before testing, make a small code change, navigate src/components/tracking/Form.js and change debug = true. This will enable you to see console logs when you submit forms.
- For testing EnableTfaModal, navigate to /account/profile and click Enable Tfa and submit the opened form. Check if you see logs for fields and form submitted.
- For testing BackupCodeModal and DisableTfaModal, create an account and Enable Tfa on the account form /account/profile page.  After enabling Tfa you should be able to see Generate Backup Codes and Disable Tfa buttons in the Two Factor Authentication section. Open both Modals and submit the respective forms and check logs for fields and form submitted.

--------------------------------------------------------------------------------
- For testing InsertSnippetModal, navigate to /templates/edit/bob-test/draft/content, click on horizontal ellipsis icon and Click on Insert Snippet menu item.

### To Do
